Dockerfile

```diff
@@ -1,58 +1,40 @@
-FROM python:3.12.9-slim AS base
+FROM python:3.12-slim AS base
 
 # Web server env var configuration
 ENV GUNICORN_WORKERS=5
 ENV GUNICORN_THREADS=8
 ENV GUNICORN_TIMEOUT=0
 
 # Allow statements and log messages to immediately appear in the logs
-ENV PYTHONUNBUFFERED True
+ENV PYTHONUNBUFFERED=True
 
-ENV APP_HOME /app
-ENV VENV_DIR .venv
+ENV APP_HOME=/app
+ENV VENV_DIR=.venv
 WORKDIR $APP_HOME
 COPY requirements.txt ./
 
 RUN apt-get update
 # install git as we need it for the direct oscrypto dependency
 # this is a temporary workaround and it should be removed once we update oscrypto to 1.3.1+
 # see: https://community.snowflake.com/s/article/Python-Connector-fails-to-connect-with-LibraryNotFoundError-Error-detecting-the-version-of-libcrypto
-RUN apt-get install -y git
+RUN apt-get install -y --no-install-recommends git
 # install libcrypt1 for IBM DB2 ibm-db package compatibility (provides libcrypt.so.1)
-RUN apt-get install -y libcrypt1
-
-# Upgrade pip globally to fix the vulnerability - VULN-510
-RUN pip install --no-cache-dir -U pip==25.0.0
+RUN apt-get install -y --no-install-recommends libcrypt1
 
 RUN python -m venv $VENV_DIR
-# Upgrade pip inside the virtual environment - VULN-510
-RUN . $VENV_DIR/bin/activate && pip install --no-cache-dir -U pip==25.0.0
 RUN . $VENV_DIR/bin/activate && pip install --no-cache-dir -r requirements.txt
 
-# VULN-423: setuptools 68.0.0 contains (CVE-2024-6345)
-RUN . $VENV_DIR/bin/activate && pip install setuptools==75.1.0
-
 # Azure database clients uses pyodbc which requires unixODBC and 'ODBC Driver 17 for SQL Server'
 # ODBC Driver 17's latest release was April, 2024. To patch vulnerabilities raised since then,
 # we have to apt-get those specific versions:
-# [VULN-602] update passwd to 1:4.13+dfsg1-1+deb12u1
-# [VULN-606] update krb5 (kerberos) to 1.20.1-2+deb12u3
-# [VULN-XXX] update libcap2 to 1:2.66-4+deb12u1
-# [VULN-613] update systemd to 252.38-1~deb12u1.
-RUN apt-get update \
-    && apt-get install -y gnupg gnupg2 gnupg1 curl apt-transport-https \
-    && curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
-    && curl https://packages.microsoft.com/config/debian/10/prod.list \
-    > /etc/apt/sources.list.d/mssql-release.list \
-    && apt-get update \
-    && ACCEPT_EULA=Y apt-get install -y msodbcsql17 unixodbc unixodbc-dev \
-    && apt-get install -y passwd=1:4.13+dfsg1-1+deb12u1 \
-    && apt-get install -y libgssapi-krb5-2=1.20.1-2+deb12u4 libkrb5-3=1.20.1-2+deb12u4 libkrb5support0=1.20.1-2+deb12u4 \
-    && apt-get install -y libcap2=1:2.66-4+deb12u2 \
-    && apt-get install -y systemd=252.39-1~deb12u1
```

Comment:
Remove pinned versions. The current Debian version has new versions of all these libs.